src/lib.rs

```diff
@@ -0,0 +1,408 @@
+                    CurrentCratePath::CurrentRustdocPath(current_rustdoc_path),
+                    &*loader,
+                    &rustdoc_cmd,
+                    name,
+                    version,
+                )?;
+
+                let success = run_check_release(&mut config, name, current_crate, baseline_crate)?;
+                vec![Ok(success)]
+            }
+            RustdocSource::Root(project_root) => {
+                let metadata = manifest_metadata(project_root)?;
+                let selected = self.scope.selected_packages(&metadata);
+                selected
+                    .iter()
+                    .map(|selected| {
+                        let manifest_path = selected.manifest_path.as_std_path();
+                        let crate_name = &selected.name;
+                        let version = &selected.version;
+
+                        let is_implied = matches!(
+                            self.scope.mode,
+                            ScopeMode::DenyList(PackageSelection {
+                                selection: ScopeSelection::Workspace,
+                                ..
+                            })
+                        ) && selected.publish == Some(vec![]);
+                        if is_implied && selected.publish == Some(vec![]) {
+                            config.verbose(|config| {
+                                config.shell_status(
+                                    "Skipping",
+                                    format_args!("{crate_name} v{version} (current)"),
+                                )
+                            })?;
+                            Ok(true)
+                        } else {
+                            config.shell_status(
+                                "Parsing",
+                                format_args!("{crate_name} v{version} (current)"),
+                            )?;
+
+                            let (current_crate, baseline_crate) = generate_versioned_crates(
+                                &mut config,
+                                CurrentCratePath::ManifestPath(manifest_path),
+                                &*loader,
+                                &rustdoc_cmd,
+                                crate_name,
+                                Some(version),
+                            )?;
+
+                            Ok(run_check_release(
+                                &mut config,
+                                crate_name,
+                                current_crate,
+                                baseline_crate,
+                            )?)
+                        }
+                    })
+                    .collect()
+            }
+            RustdocSource::Revision(_, _) | RustdocSource::Version(_) => todo!(),
+        };
+        let success = all_outcomes
+            .into_iter()
+            .fold_ok(true, std::ops::BitAnd::bitand)?;
+
+        Ok(Report { success })
+    }
+}
+
+pub struct Report {
```

Comment:
```suggestion
#[non_exhaustive]
#[derive(Debug)
pub struct Report {
```
I know `#[non_exhaustive]` is redundant here at the moment, but better safe than sorry. We don't want to end up in a situation where we've accidentally removed the last private field without realizing, and we're then locked into a constructible exhaustive struct until the next major version.

One day `cargo-semver-checks` will warn about "semver-regrettable" situations like that.